File: app/src/main/java/com/khinthirisoe/bakingapp/ui/BakingWidgetRemoteViewsService.java
package com.khinthirisoe.bakingapp.ui;

import android.content.Intent;
import android.util.Log;
import android.widget.RemoteViewsService;

public class BakingWidgetRemoteViewsService extends RemoteViewsService {
    private static final String TAG = "WidgetService";

    @Override
    public RemoteViewsFactory onGetViewFactory(Intent intent) {
        Log.d(TAG, "onGetViewFactory: " + "Service called");
        return new BakingWidgetRemoteViewsFactory(this.getApplicationContext(), intent);
    }
}

File: app/src/main/java/com/khinthirisoe/bakingapp/ui/widget/BakingWidgetRemoteViewsService.kt
package com.khinthirisoe.bakingapp.ui.widget

import android.content.Intent
import android.widget.RemoteViewsService

class BakingWidgetRemoteViewsService : RemoteViewsService() {

    override fun onGetViewFactory(intent: Intent): RemoteViewsService.RemoteViewsFactory {
        return BakingWidgetRemoteViewsFactory(this.applicationContext, intent)
    }
}
